docs/src/app/(public)/(content)/react/utils/use-render/demos/render-callback/css-modules/index.tsx

```diff
@@ -0,0 +1,55 @@
+'use client';
+import * as React from 'react';
+import { useRender } from '@base-ui-components/react/use-render';
+import styles from './index.module.css';
+
+type CounterState = {
+  odd: boolean;
+};
+
+type CounterProps = {
+  className?: string;
+  render?: useRender.RenderProp<CounterState>;
+};
+
+function Counter(props: CounterProps) {
+  const { render, className, ...otherProps } = props;
+  const [count, setCount] = React.useState(0);
+  const odd = count % 2 === 1;
+  const state = React.useMemo(() => ({ odd }), [odd]);
+
+  const { renderElement } = useRender({
+    render: render ?? <button type="button" />,
+    state,
+    props: {
+      ...otherProps,
+      className: `${className} ${styles.Button}`,
+      type: 'button',
+      children: (
+        <React.Fragment>
+          Counter: <span>{count}</span>
+        </React.Fragment>
+      ),
+      onClick: () => setCount((prev) => prev + 1),
+      'aria-label': `Count is ${count}, click to increase.`,
+    },
+  });
+
+  return renderElement();
+}
+
+export default function ExampleCounter() {
+  return (
+    <Counter
+      render={(props, state) => (
+        <button
+          {...props}
+          className={`${state.odd ? styles.odd : styles.even} ${props.className}`}
+        >
+          {props.children}
+          <span className={styles.suffix}>{state.odd ? '👎' : '👍'}</span>
```

Comment:
I'd choose between the class name and the emoji. One of them is enough. We can leave the emoji and remove the class name to simplify the demo.